testbed/testbed/load_generator.go

```diff
@@ -210,22 +208,41 @@ func (ps *ProviderSender) generate() {
 			defer workers.Done()
 			t := time.NewTicker(time.Second / time.Duration(ps.options.DataItemsPerSecond/ps.options.ItemsPerBatch/numWorkers))
 			defer t.Stop()
+
+			var (
+				prevErr error
```

Comment:
Is `prevErr` supposed to hold all previous errors, just the last error, or something else? I'm trying to better understand the point here before making more suggestions.

Reply:
As per my understanding of the code,

1. Before this PR, the `prevErr` was supposed to hold the last error that occurred during load generation in any of the goroutines -- not sure if this behavior was intended. Also, this is what introduced the race condition that the PR is trying to solve.
2. After this PR, the `prevErr` holds the last error for each goroutine generating the load.

Alternatively, we can get rid of the `prevErr` completely and build a `onceLogger` that can use some probabilistic data structure (like bloom filter) to track what has happened before and only log new errors.